### PR TITLE
Create servicelists using tvheadend server API

### PR DESCRIPTION
**Pull request by reuab**
I've added the possibility to create a servicelist from tvheadend servers without copying the configuration files. I have also updated the README file in the source-input folder, the linefeed and the exit code at the end of both scripts.

If you like the modification, i'd love to see it merged into your repo.
Thanks for the your work, btw.

### CODE REVIEW

**Comment by picons** [comment]
Will look at it some more later, but has a big chance for being merged...

    and it is even about 20% faster
Is that because of my hacked together code? :smile:

File: 1-build-servicelist.sh [hunk 1]
```diff
@@ -116,9 +116,86 @@ if [[ -d $location/build-input/tvheadend ]]; then
...
+
+    # ...replace default credentials by those configured in file
+    source "$location/build-input/tvheadend.serverconf"
+
+    # ...set file name for the service list to generate
+    file="$location/build-output/servicelist-tvh-server-$style"
+    tempfile="$temp/$(echo $RANDOM)"
+
+    # ...if an old file with that name exists, delete it
+    [[ -f "${file}" ]] && rm "${file}"
```
Comment (picons) [inline 1-build-servicelist.sh]: Do we really need this? Trying to keep the number of code lines to the bare minimum....

File: 1-build-servicelist.sh [hunk 2]
```diff
@@ -116,9 +116,86 @@ if [[ -d $location/build-input/tvheadend ]]; then
...
+            if [[ -z $logo_snp ]]; then logo_snp="--------"; fi
+            echo -e "$serviceref\t$channelname\t$serviceref_id=$logo_srp\t$snpname=$logo_snp" >> "$tempfile"
+        else
+            echo -e "$serviceref\t$channelname\t$serviceref_id=$logo_srp" >> "$tempfile"
+        fi
+      done
+
+      sort -t $'\t' -k 2,2 "$tempfile" | sed -e 's/\t/§|/g' | column -t -s $'§' | sed -e 's/|/  |  /g' > "$file"
+      rm "$tempfile"
+      echo "TVHeadend server \"${TVH_HOST}\" channels exported to ${file}"
```
Comment (picons) [inline 1-build-servicelist.sh]: Could you also keep the echo output the same as with the rest....not too much info also, I think everybody knows for which host they are running the script....I just want to keep as much of it the same for the different parts, if that makes any sense.

File: build-input/README.md [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+TVH_PASS=""
+```
+
+You need only those values in your file which are different from the defaults.
+For most people a file with, a single host name or host IP address is sufficient:
+```sh
+TVH_HOST="my.tvheadend.server"
+```
+
+If your running Tvheadend on the same machine, even an empty file (defaulting to `localhost`) might do it.
```
Comment (picons) [inline build-input/README.md]: should be `you're` I think. And it `should do it`, if it `might do it`, we've got a problem....

File: build-input/README.md [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+TVH_PORT="9981"
+
+# tvheadend user name
+TVH_USER=""
+
+# tvheadend password of above user
+TVH_PASS=""
+```
+
+You need only those values in your file which are different from the defaults.
```
Comment (picons) [inline build-input/README.md]: perhaps `you only need` reads better, no native English speaker so just guessing here.

File: build-input/README.md [hunk 5]
```diff
@@ -1,3 +1,43 @@
...
+1) Use the server generated configuration files by copying your `tvheadend` configuration folder,
+probably located in `/home/hts/.hts` on your server into this folder.
+
+2) Use the servers API and directly ask the server about all channels by creating a file
+called `tvheadend.serverconf`.
+
+
+The first option has the advantage to work even without a running server.
+
+The advantage of the second option is that you dont have to copy files around,
```
Comment (picons) [inline build-input/README.md]: don't

File: 1-build-servicelist.sh (hunk 1 above)
Comment (reuab) [inline 1-build-servicelist.sh]: Yes, i understand ..wise decision!
I am really not sure if i have to remove an already existing file (line 149). Other sections (like VDR etc. don't delete existing files.
However, the rest IS required ;-)

File: 1-build-servicelist.sh (hunk 2 above)
Comment (reuab) [inline 1-build-servicelist.sh]: I just wanted different section to produce different output. I thought, this way it is easier to spot problems.
But i will keep it in mind. I also can modify it to be the same, if you like.

File: build-input/README.md (hunk 3 above)
Comment (reuab) [inline build-input/README.md]: :-D ..thx for fixing this!
Sorry, i was a bit in a hurry. And "unfortunately" my native tongue is german ..actually even worse swiss-german :-)

**Comment by picons** [comment]
It's already been merged and modified by me....but over at picons/picons the show continues.

**Pull request merged**